src/main/java/com/code_intelligence/jazzer/driver/FuzzTargetFinder.java

```diff
@@ -143,28 +147,31 @@ private static Optional<FuzzTarget> findFuzzTargetByAnnotation(Class<?> clazz) {
           parameter.getName(), method.getName(), clazz.getName()));
     }
 
-    if (clazz.getDeclaredConstructors().length != 1) {
-      throw new IllegalArgumentException(String.format(
-          "Classes containing a method annotated with @FuzzTest must declare exactly one constructor, got multiple in %s",
-          clazz.getName()));
-    }
-    Constructor<?> constructor = clazz.getDeclaredConstructors()[0];
-    // JUnit 5 supports injected constructor parameters, but we don't.
-    if (constructor.getParameterCount() > 0) {
-      throw new IllegalArgumentException(String.format(
-          "The constructor of a class containing a method annotated with @FuzzTest must take no parameters, got a non-zero number in %s",
-          clazz.getName()));
+    Callable<Object> newInstance;
+    if (FuzzTargetInstanceHolder.fuzzTargetInstance != null) {
```

Comment:
Could you clarify this check? If I see it correctly, `FuzzTargetInstanceHolder.fuzzTargetInstance` is only set by `FuzzTestExecutor` when fuzzing from JUnit. In which cases is the fuzz test instance not `null` when finding the fuzz test in the driver?

Reply:
The `FuzzTargetRunner` is also used by the JUnit executor. That is the case where this is not `null`, but the other branch will go away when we use the JUnit launcher API. I added comments explaining that.

Reply:
Thanks for the clarification. I've overseen the fact that control flows through the static block and the `newInstance` lambda.

Reply:
Yes, it is confusing. I hope that we can get rid of most of this logic when we migrate to the JUnit launcher.